### PR TITLE
Add leaf index as a key on the log line

### PR DESCRIPTION
**Pull request by nambrot**
This allows GCP to correctly filter for it in stackdriver logging

### CODE REVIEW

**Review by tkporter — APPROVED** [review]
Should we change the log message to just be `Message processing failed` ? Feels unnecessary to have it once and it's more useful when dealing with structured logs if the message is uniform

File: rust/agents/relayer/src/msg/serial_submitter.rs [hunk 1]
```diff
@@ -228,7 +228,7 @@ impl SerialSubmitter {
                 info!(msg=?msg, "Message processed");
             }
             Err(e) => {
-                info!(msg=?msg, "Message processing failed: {}", e);
```
Comment (mattiecnvr) [inline rust/agents/relayer/src/msg/serial_submitter.rs:231]: Please, don't put formatted data into the structured message. It makes it very hard to filter. It is much better if it is added as a separate field to the object.

Comment (mattiecnvr) [reply rust/agents/relayer/src/msg/serial_submitter.rs:231]: (I realize it was that way before, but great time to fix)

File: rust/agents/relayer/src/msg/serial_submitter.rs [hunk 2]
```diff
@@ -228,7 +228,7 @@ impl SerialSubmitter {
...
+                info!(msg=?msg, leaf_index=msg.leaf_index, error=?e, "Message processing failed: {}", e);
```
Comment (nambrot) [inline rust/agents/relayer/src/msg/serial_submitter.rs:231]:
```suggestion
                info!(msg=?msg, leaf_index=msg.leaf_index, error=?e, "Message processing failed");
```

File: rust/agents/relayer/src/msg/serial_submitter.rs (hunk 1 above)
Comment (nambrot) [reply rust/agents/relayer/src/msg/serial_submitter.rs:231]: Agreed and moved it out

**Pull request merged**